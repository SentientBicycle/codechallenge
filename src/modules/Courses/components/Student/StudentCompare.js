import React, { Component } from "react";
import style from "../../Courses.module.css";

class StudentCompare extends Component {
	render() {
		return (
			<div className={style["card-compare"]}>
				<div className={style["compare-content-left"]}>
					{this.props.student.first_name}{" "}
					{this.props.student.last_name}
				</div>
				<div className={style["compare-content-right"]}>
					{this.props.student.currentClassScore}
				</div>
			</div>
		);
	}
}

export default StudentCompare;
