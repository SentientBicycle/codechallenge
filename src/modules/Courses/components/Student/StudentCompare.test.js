import React from 'react';
import ReactDOM from 'react-dom';
import StudentCompare from './StudentCompare'
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const props = {
	"student": {
		"first_name": "Chris",
		"last_name": "Goodwin",
		"currentClassScore": 55
	}
}

describe("StudentCompare", () =>{
	it('Renders properly', () => {
	  const course = shallow(<StudentCompare {...props}/>);
	  expect(course).toMatchSnapshot();
	});
});
