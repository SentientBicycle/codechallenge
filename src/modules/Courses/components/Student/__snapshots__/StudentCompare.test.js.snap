// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCompare Renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StudentCompare
    student={
      Object {
        "currentClassScore": 55,
        "first_name": "Chris",
        "last_name": "Goodwin",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Chris
           
          Goodwin
        </div>,
        <div>
          55
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Chris",
            " ",
            "Goodwin",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          "Chris",
          " ",
          "Goodwin",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 55,
          "className": undefined,
        },
        "ref": null,
        "rendered": 55,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Chris
             
            Goodwin
          </div>,
          <div>
            55
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Chris",
              " ",
              "Goodwin",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            "Chris",
            " ",
            "Goodwin",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 55,
            "className": undefined,
          },
          "ref": null,
          "rendered": 55,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
